packages/koa-metrics/src/index.ts

```diff
@@ -0,0 +1,56 @@
+import {StatsD} from 'hot-shots';
+import {Context} from 'koa';
+
+import {tagsForRequest, tagsForResponse} from './tags';
+import {initTimer} from './timing';
+import {instrumentContentLength} from './content';
+
+export interface Options {
+  prefix: string;
+  host: string;
+  skipInstrumentation?: boolean;
+}
+
+export default function metrics({
+  prefix,
+  host,
+  skipInstrumentation = false,
+}: Options) {
+  const statsdHost = host.split(':')[0];
+  const statsdPort = parseInt(host.split(':')[1], 10);
+
+  return async function statsdMiddleware(ctx: Context, next: Function) {
+    const client = new StatsD({
+      host: statsdHost,
+      port: statsdPort,
+      prefix,
+      globalTags: tagsForRequest(ctx),
+    });
+
+    let timer;
+
+    if (!skipInstrumentation) {
+      timer = initTimer(client, ctx);
+    }
+
+    // allow later middleware to add metrics
+    ctx.state.statsd = client;
+
+    try {
+      await next();
+    } finally {
+      const responseClient = client.childClient({
+        globalTags: tagsForResponse(ctx),
```

Comment:
I wonder if these tags should be somehow added to metrics logged within the `await next()` call? thoughts @lemonmade ? I could perhaps make a wrapper around the statsd interface that looks at ctx?

Reply:
I don't understand how you think that would be possible since these tags are probably not set until the very end of the middleware chain for the most part